tests/js/legendTests.js

```diff
@@ -0,0 +1,192 @@
+  }
+];
+
+    gpii.tests.chartAuthoring.mouseOverListener = function (data, i, that) {
+        that.mouseOverListenerCalled = true;
+    };
+
+    // convenience function for easing testing of colors (jquery returns only RGB)
+    // Based off http://stackoverflow.com/questions/4262417/jquery-hex-to-rgb-calculation-different-between-browsers
+    gpii.tests.chartAuthoring.hexToRGB = function (hexStr){
+        // note: hexStr should be #rrggbb
+        var hex = parseInt(hexStr.substring(1), 16);
+        var r = (hex & 0xff0000) >> 16;
+        var g = (hex & 0x00ff00) >> 8;
+        var b = hex & 0x0000ff;
+        return "rgb("+r+", "+g + ", " +b + ")";
+    };
+
+    gpii.tests.chartAuthoring.testMouseOverListener = function (that) {
+      jqUnit.assertFalse("The mouseover listener for legend rows has not been triggered", that.mouseOverListenerCalled);
+      var oneD3Row = that.jQueryToD3($(that.locate("row")[0]));
+      oneD3Row.on("mouseover")();
+      jqUnit.assertTrue("The mouseover listener for legend rows has been triggered", that.mouseOverListenerCalled);
+    };
+
+    gpii.tests.chartAuthoring.validateLegend = function (that) {
+        var table = that.locate("table");
+
+        // Test the legend creation
+        jqUnit.assertNotEquals("The TABLE element for the legend is created with the proper selector", 0, table.length);
+
+        jqUnit.assertEquals("A TR element has been created for each value in the dataset, with proper selectors", that.model.dataSet.length, that.locate("row").length);
+
+        var d3ColorCells = that.jQueryToD3($(that.locate("legendColorCell")));
+        d3ColorCells.each(function (d) {
+
+            jqUnit.assertEquals("The data colors are filled correctly in the legend", gpii.tests.chartAuthoring.hexToRGB(d.color), ($(this).css("background-color")));
+        });
+
+        var d3LabelCells = that.jQueryToD3($(that.locate("labelCell")));
+        d3LabelCells.each(function (d) {
+            jqUnit.assertEquals("The data labels are applied correctly in the legend", d.label, ($(this).html()));
+        });
+
+        var d3ValueCells = that.jQueryToD3($(that.locate("valueCell")));
+
+        d3ValueCells.each(function (d) {
+            jqUnit.assertEquals("The data values are applied correctly in the legend", d.value, ($(this).html()));
+        });
+
+    };
+
+    gpii.tests.chartAuthoring.testLegend = function (that) {
+      // Legend is created from dataset
+
+      gpii.tests.chartAuthoring.validateLegend(that);
+
+      gpii.tests.chartAuthoring.testMouseOverListener(that);
+
+      // Legend is redrawn when data set changes
+
+      // Item added to dataset
+
+      that.applier.change("dataSet", gpii.tests.chartAuthoring.objectArrayAdd);
+      gpii.tests.chartAuthoring.validateLegend(that);
+
+      // Item removed from dataset
+
+      that.applier.change("dataSet", gpii.tests.chartAuthoring.objectArrayRemove);
+      gpii.tests.chartAuthoring.validateLegend(that, false);
```

Comment:
gpii.tests.chartAuthoring.validateLegend.validateLegend() doesn't accept the second argument.